File: gaia-sdk-python/gaia_sdk/graphql/response/type/UpdatedRoleImpulse.py

from gaia_sdk.graphql.response.type.Role import Role

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class UpdatedRoleImpulse:
    """
    Impulse which indicates the result of a update role impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    @property
    def data(self) -> Role:
        return Role(self.dictionary.get("data"))

File: gaia-sdk-javascript/src/graphql/response/type/UpdatedRoleImpulse.ts

import {Role} from "./Role";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a update role impulse
*/
export interface UpdatedRoleImpulse {
    id?:Uuid, 
    data?:Role
}